oauth2_http/java/com/google/auth/oauth2/S2A.java

```diff
@@ -0,0 +1,107 @@
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.client.util.GenericData;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ServiceLoader;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utilities to fetch the S2A (Secure Session Agent) address from the mTLS configuration.
+ *
+ * <p>mTLS configuration is queried from the MDS MTLS Autoconfiguration endpoint.
+ */
+@ThreadSafe
+public final class S2A {
+  public static final String DEFAULT_METADATA_SERVER_URL = "http://169.254.169.254";
```

Comment:
It seems like getting metadata server address is already implemented here: https://github.com/googleapis/google-auth-library-java/blob/dde6876154d6e7ff43dbbf7b75c5205eadd679a5/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L467

should we reuse that definition?

Reply:
Done. Used ComputeEngineCredentials.getMeadataServerUrl()